bayesbridge/design_matrix/abstract_matrix.py

```diff
@@ -63,12 +70,16 @@ def toarray(self):
         """ Returns a 2-dimensional numpy array. """
         pass
 
+    @staticmethod
+    def is_cupy_sparse(X):
+        return isinstance(X, cpx.scipy.sparse.spmatrix) if cp and cpx else False
+
     @staticmethod
     def remove_intercept_indicator(X):
-        if sp.sparse.issparse(X):
-            col_variance = np.squeeze(np.array(
-                X.power(2).mean(axis=0) - np.power(X.mean(axis=0), 2)
-            ))
+        squeeze, array, power = (cp.squeeze, cp.array, cp.power) if \
+            AbstractDesignMatrix.is_cupy_sparse(X) else (np.squeeze, np.array, np.power)
+        if sp.sparse.issparse(X) or AbstractDesignMatrix.is_cupy_sparse(X):
+            col_variance = squeeze(array(X.power(2).mean(axis=0) - power(X.mean(axis=0), 2)))
```

Comment:
Would the following work (w/o `{np, cp}.power()`)?
```suggestion
            col_variance = squeeze(array(X.power(2).mean(axis=0) - X.mean(axis=0).power(2)))
```

Reply:
I don't think so, `power()` doesnt appear to be a method for matrices (for np and cp).
```
In [10]: X = np.ones((5,5))

In [11]: X.mean(axis=0).power(2)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-2fff9e220fa4> in <module>
----> 1 X.mean(axis=0).power(2)

AttributeError: 'numpy.ndarray' object has no attribute 'power'
```

Reply:
Ah, silly me. Thanks, good to know.